### PR TITLE
fixed vertical layout

### PR DESCRIPTION
**Pull request by ghost**
## What does it do?
Makes page look better on devices

## Why the change?
Better vertical layout support

## How can this be tested?
<How is the best way to test this?>

## Where to start code review?
<Point us to a good place to start reviewing>

## Relevant tickets?
<Please link any relevant tickets>

## Questions?
<Ask us anything!>

### CODE REVIEW

File: src/js/UI.js [hunk 1]
```diff
@@ -146,7 +146,6 @@ export default class UI {
     }
 }
 
-Split(['#left_pane', '#right_pane'], {
-    gutterSize: 8,
-    cursor: 'col-resize'
-  })
\ No newline at end of file
+if (window.innerWidth > 960) {
```
Comment (DoumanAsh) [inline src/js/UI.js]: Move it to `src/main.js` in load event handler

File: src/js/dragging.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
+export function drag() {
+    let leftPane = document.getElementById('left_pane')
```
Comment (DoumanAsh) [inline src/js/dragging.js]: Initialize `leftPane` and two next variables outside of function itself.

Use all upper case with snake case for naming constants

File: src/js/dragging.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    let rightPane = document.getElementById('right_pane')
+    let separator = document.getElementById('separator')
+
+    separator.style.cursor = 'move'
+
+    let leftLimit = 10
```
Comment (DoumanAsh) [inline src/js/dragging.js]: These variables are constants, use `const` and snake case for all namings here and everywhere else

File: src/js/dragging.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            dragging = false
+        }
+
+        separator.addEventListener('mousedown', dragstart)
+        window.addEventListener('mousemove', function(e) {dragmove(e)})
+        window.addEventListener('mouseup', dragend)
+        
+    }
+
+    sdrag(function (el, pageX, startX, pageY, startY, fix) {
```
Comment (DoumanAsh) [inline src/js/dragging.js]: Please declare function in sdrag separately to make it easier to read.

File: src/js/dragging.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
+export function drag() {
+    let leftPane = document.getElementById('left_pane')
+    let rightPane = document.getElementById('right_pane')
+    let separator = document.getElementById('separator')
+
+    separator.style.cursor = 'move'
```
Comment (DoumanAsh) [inline src/js/dragging.js]: Use `col-resize` for horizontal layout.

It would be better to move styling to CSS

File: src/js/dragging.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        let dragging = false
+
+        function dragstart() {
+            dragging = true
+        }
+    
+        function dragmove(e) {
+            if (dragging) {
+                let fix = {}
+                    onDrag && onDrag(el, e.pageX, startX, e.pageY, startY, fix)
```
Comment (DoumanAsh) [inline src/js/dragging.js]: Please use `if` to for check instead of `onDrag && ...`
Also correct tabulation of this and next lines

File: src/js/dragging.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+                    if (direction !== 'vertical') {
```
Comment (DoumanAsh) [inline src/js/dragging.js]: Use constants for `vertical` and `horizontal`

File: src/js/dragging.js [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+    let leftPane = document.getElementById('left_pane')
+    let rightPane = document.getElementById('right_pane')
+    let separator = document.getElementById('separator')
+
+    separator.style.cursor = 'move'
+
+    let leftLimit = 10
+    let rightLimit = 90
+
+    function sdrag(onDrag, onStop, direction) {
```
Comment (DoumanAsh) [inline src/js/dragging.js]: Function should have more meaningful name `perform_drag`

File: src/js/dragging.js [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+                    }
+            }
+        }
+    
+        function dragend() {
+            dragging = false
+        }
+
+        separator.addEventListener('mousedown', dragstart)
+        window.addEventListener('mousemove', function(e) {dragmove(e)})
```
Comment (DoumanAsh) [inline src/js/dragging.js]: Pass `dragmove` as it is

File: src/js/dragging.js [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+                        }
+                        if (!('skipY' in fix)) {
+                            el.style.top = (pageY - startY) + 'px'
+                        }
+                    }
+            }
+        }
+    
+        function dragend() {
+            dragging = false
```
Comment (DoumanAsh) [inline src/js/dragging.js]: maybe instead of relying on flag, we could add/remove listener for `mousemove` event?

File: src/js/dragging.js [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+        separator.addEventListener('mousedown', dragstart)
+        window.addEventListener('mousemove', function(e) {dragmove(e)})
+        window.addEventListener('mouseup', dragend)
+        
+    }
+
+    sdrag(function (el, pageX, startX, pageY, startY, fix) {
+        fix.skipX = true
+
+        if (pageX < window.innerWidth * leftLimit / 100) {
```
Comment (DoumanAsh) [inline src/js/dragging.js]: Compute `window.innerWidth * leftLimit / 100` once by saving it to variable

File: src/js/dragging.js [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+            pageX = window.innerWidth * leftLimit / 100
+            fix.pageX = pageX
+        }
+        if (pageX > window.innerWidth * rightLimit / 100) {
```
Comment (DoumanAsh) [inline src/js/dragging.js]: Compute `pageX > window.innerWidth * rightLimit / 100` once by saving it to variable

File: src/js/UI.js (hunk 1 above)
Comment (Leandros) [inline src/js/UI.js]: This value should be somewhere configurable, and be kept in sync to what's in the media-queries. To easy to mistakenly change it, and do not update the queries, or the other way around.

File: src/js/dragging.js (hunk 6 above)
Comment (Leandros) [inline src/js/dragging.js]: Erroneous indentation?

File: src/js/dragging.js (hunk 4 above)
Comment (Leandros) [inline src/js/dragging.js]: Please use descriptive function / variable names.

File: src/styles/fonts.sss [hunk 13]
```diff
@@ -26,4 +26,3 @@
     src:
         url("../fonts/Source_Sans_Pro/SourceSansPro-SemiBold.ttf")
         format("truetype")
-
```
Comment (Leandros) [inline src/styles/fonts.sss]: Please configure your editor to not remove the last newline in a file, it creates unnecessary whitespace diff.

File: src/js/UI.js [hunk 14]
```diff
@@ -3,7 +3,7 @@
 import {compile} from './compiler.js'
 import {toggle_class} from './utils.js'
 import {create_editor} from './editor.js'
-import Split from 'split.js'
```
Comment (Leandros) [inline src/js/UI.js:6]: If you remove split.js, please also remove it from the dependencies.

File: src/styles/layout.sss [hunk 15]
```diff
@@ -41,19 +41,16 @@
 .main
     flex: 5 0 60%
     display: flex
-    box-sizing: border-box
+    overflow: scroll
```
Comment (DoumanAsh) [inline src/styles/layout.sss:44]: Why you need scroll for main window?
Scrolling should be inside panes themself

File: src/styles/layout.sss [hunk 16]
```diff
@@ -91,7 +88,6 @@
 .main__compiler__output
     flex: 3 0 30%
     max-height: 25%
-    overflow: auto
```
Comment (DoumanAsh) [inline src/styles/layout.sss:94]: Why you removed overflow?

File: src/js/dragging.js [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+        let cur = pageX / window.innerWidth * 100
+        if (cur < 0) {
+            cur = 0
+        }
+        if (cur > window.innerWidth) {
+            cur = window.innerWidth
+        }
+
+        let right = (100-cur-2)
+        leftPane.style.width = cur + '%'
```
Comment (DoumanAsh) [inline src/js/dragging.js]: Current calculations are only for horizontal layout. You should consider using different event handlers for vertical/horizontal layouts to simplify code.
And on each layout change you need to unset `width`/`height` otherwise you break layout.

**Review by DoumanAsh — CHANGES_REQUESTED** [review]
I feel that it might be better to consider using Split as it supports option of layout. 

Could you maybe investigate if it is possible just reconfigure Split.js on fly?

I would continue review after you'll rework your code.

My suggestion, if not using Split.js, is using two even handlers for vertical/horizontal layout.
It should be chosen on start by checking viewport size.
You will also need to add handler for `resize` which basically should determine current layout and change, if necessary, event handler

File: src/styles/layout.sss (hunk 15 above)
Comment (Leandros) [reply src/styles/layout.sss:44]: Scrolling should be inside pane, only in horizontal layout, not in vertical layout. Vertically, the whole content should expand the pane.

Comment (DoumanAsh) [reply src/styles/layout.sss:44]: Then it should be under `@media screen and (max-width: 960px)` below.
But overflow should be `auto` instead of `scroll.
And most likely we'd need to use overflow-y instead of just overflow.

**Comment by DoumanAsh** [comment]
@sergebraun In case you'd like to discuss something, while i'm banned from slack, you can reach me through skype `coldgloom`
